packages/playground/data-liberation/src/import/WP_Topological_Sorter.php

```diff
@@ -212,111 +249,243 @@ public function delete_session( $session_id ) {
 	}
 
 	/**
-	 * Map a category to the index.
+	 * Called by 'wxr_importer_pre_process_*' filters. This populates the entity
+	 * object with the mapped IDs.
 	 *
-	 * @param int $byte_offset The byte offset of the category.
-	 * @param array $data The category data.
+	 * @param array $data The data to map.
+	 * @param int|null $id The ID of the element.
+	 * @param int|null $additional_id The additional ID of the element.
 	 */
-	public function map_category( $byte_offset, $data ) {
-		global $wpdb;
+	public function filter_wxr_importer_pre_process( $data, $id = null, $additional_id = null ) {
+		$current_session = $this->current_session;
+		$current_filter  = current_filter();
+		$types           = array(
+			'wxr_importer_pre_process_comment'      => 'comment',
+			'wxr_importer_pre_process_comment_meta' => 'comment_meta',
+			'wxr_importer_pre_process_post'         => 'post',
+			'wxr_importer_pre_process_post_meta'    => 'post_meta',
+			'wxr_importer_pre_process_term'         => 'term',
+		);
+
+		if ( ! $current_filter || ! array_key_exists( $current_filter, $types ) ) {
+			_doing_it_wrong(
+				__METHOD__,
+				'This method should be called by the wxr_importer_pre_process_* filters.',
+				'1.0.0'
+			);
 
-		if ( empty( $data ) ) {
 			return false;
 		}
 
-		$category_parent = null;
-
-		if ( array_key_exists( 'parent', $data ) && '' !== $data['parent'] ) {
-			$category_parent = $data['parent'];
-		}
+		return $this->get_mapped_element( $types[ $current_filter ], $data, $id, $additional_id );
+	}
 
-		$wpdb->insert(
-			self::get_table_name(),
-			array(
-				'session_id'   => $this->current_session,
-				'element_type' => self::ELEMENT_TYPE_CATEGORY,
-				'element_id'   => (string) $data['term_id'],
-				'parent_id'    => $category_parent,
-				'byte_offset'  => $byte_offset,
-				// Items with a parent has at least a sort order of 2.
-				'sort_order'   => $category_parent ? 2 : 1,
-			)
+	/**
+	 * Called by 'wxr_importer_processed_*' actions. This adds the entity to the
+	 * sorter table.
```

Comment:
`public function action_wxr_importer_processed` reflects the context where the method is called. Can we, instead, reflect the intended outcome of that method? As in `store_imported_entity` or so. 

I know nitpicking names is annoying, but I think it's worth it in the long run. It makes the difference between encoding the intentions directly in the system and building one where a human needs to recover the intention by jumping around the codebase.